src/main/java/jenkins/plugins/openstack/compute/JCloudsSlave.java

```diff
@@ -214,6 +214,33 @@ private static Openstack getOpenstack(String cloudName) {
         return JCloudsCloud.getByName(cloudName).getOpenstack();
     }
 
+    public JCloudsSlaveTemplate getTemplate() {
+        JCloudsCloud cloud = JCloudsCloud.getByName(cloudName);
+        Server server = cloud.getOpenstack().getServerById(getServerId());
+        String templateName = server.getMetadata().get(JCloudsSlaveTemplate.OPENSTACK_TEMPLATE_NAME_KEY);
+        return cloud.getTemplate(templateName);
```

Comment:
No need to invoke OS calls. Both cloud and template names can be found in `id` field. Note template can be null and none of them is guaranteed to be still configured.